packages/shared/src/components/fields/PasswordField.tsx

```diff
@@ -45,20 +52,32 @@ export function PasswordField({
     ? `Password need a minimum length of ${props.minLength}`
     : passwordStrengthStates[passwordStrengthLevel].label;
   const hint = !hasUserAction ? null : userActionHint;
-  const shouldShowStrength = showStrength && hasUserAction;
+  const shouldShowStrength =
+    !!value && showStrength && hasUserAction && indicateStrength;
   const Icon = useType === 'password' ? EyeIcon : EyeCancelIcon;
 
+  const onIndicateStrength = () => {
+    if (!indicateStrength) {
+      setIndicateStrength(true);
+    }
+  };
+
+  const onChange = (input: string) => {
+    setPasswordStrengthLevel(passwordStrength(input).id);
+    setValue(input);
+  };
+
   return (
     <TextField
       {...props}
       autoComplete="off"
       type={useType}
-      valueChanged={(value) =>
-        setPasswordStrengthLevel(passwordStrength(value).id)
-      }
+      valueChanged={onChange}
       leftIcon={<LockIcon />}
-      hint={showStrength ? hint : props.hint}
+      hint={showStrength && indicateStrength ? hint : props.hint}
```

Comment:
I think this is becoming a little bit messy with mulitple states indicating if it should show or not?
Or maybe I'm misreading it.

But do we really need to have it so complex?
Why don't we simply always show the strength once inputted or validating?

Reply:
I agree, this was the previous work before we had that condition. Let me refactor this one.